### PR TITLE
docs: Improve developers installation guide

### LINKED ISSUE #511: simplify the developers guide
The developers guide currently include instructions that are duplicated in the installation guides, so drop these details and point to the installation guides to avoid duplication.

### PR DESCRIPTION
**Pull request by jodh-intel**
Remove all instructions that are duplicated in the installation guides
and require developers to install via the installation instructions as
a prerequisite.

Also fixup some formatting (headings and indents).

Fixes #511.

Signed-off-by: James O. D. Hunt <james.o.hunt@intel.com>

### CODE REVIEW

**Comment by jodh-intel** [comment]
Hi @iphutch - please could you review?

File: docs/developers-clear-containers-install.md [hunk 1]
```diff
@@ -18,122 +34,70 @@ process.
...
-$ go get -d github.com/clearcontainers/runtime
-$ go get -d github.com/clearcontainers/proxy
-$ git clone https://github.com/clearcontainers/shim $GOPATH/src/github.com/clearcontainers/shim
-$ go get -d github.com/clearcontainers/tests
-```
+   ```bash
+   $ go get -d github.com/clearcontainers/runtime
+   $ go get -d github.com/clearcontainers/proxy
+   $ git clone https://github.com/clearcontainers/shim $GOPATH/src/github.com/clearcontainers/shim
+   $ go get -d github.com/clearcontainers/tests
```
Comment (devimc) [inline docs/developers-clear-containers-install.md]: missing agent?

File: docs/developers-clear-containers-install.md [hunk 2]
```diff
@@ -5,7 +5,23 @@ instructions for any developers eager to try Clear Containers 3.0 and who
...
+
+  * [Fedora](fedora-installation-guide.md)
+  * [Ubuntu](ubuntu-installation-guide.md)
+
+The instructions will install all packaged components, plus docker,
+the hypervisor and the Clear Containers image and kernel.
+
+After validating it is possible to create a Clear Container (the final
+step in the installation guides), progress to installing the desired
+component(s) you wish to develop.
```
Comment (iphutch) [inline docs/developers-clear-containers-install.md]: This sentence is hard to follow. "Progress to" can be interpreted as 'follow instructions up to', or 'next, install' - I suggest we break it into two parts following the closing parenthesis and replace "progress to".

File: docs/developers-clear-containers-install.md [hunk 3]
```diff
@@ -5,7 +5,23 @@ instructions for any developers eager to try Clear Containers 3.0 and who
...
+## Initial setup
+
+The recommended way to create a development environment is to first install the
+packaged versions of the Clear Containers components to create a working
+system:
+
+  * [Fedora](fedora-installation-guide.md)
+  * [Ubuntu](ubuntu-installation-guide.md)
+
+The instructions will install all packaged components, plus docker,
```
Comment (iphutch) [inline docs/developers-clear-containers-install.md]: state which instructions you are referring to:
The installation guide instructions OR The instructions in this guide

File: docs/developers-clear-containers-install.md [hunk 4]
```diff
@@ -5,7 +5,23 @@ instructions for any developers eager to try Clear Containers 3.0 and who
...
 process.
 
-## Requirements
+## Initial setup
+
+The recommended way to create a development environment is to first install the
+packaged versions of the Clear Containers components to create a working
+system:
+
+  * [Fedora](fedora-installation-guide.md)
```
Comment (iphutch) [inline docs/developers-clear-containers-install.md]: Fedora and Ubuntu must both have \* following them.

File: docs/developers-clear-containers-install.md (hunk 1 above)
Comment (jodh-intel) [reply docs/developers-clear-containers-install.md]: I thought about that. The problem with adding the agent on this PR (which is just to update what we already have) is that we'll then need to explain how to install it into an image (it won't be just `sudo make install` like the rest) and I don't think even the osbuilder docs currently cover that.

I think we probably need a doc in the agent repo to explain so that _this_ document can reference it.

@sameo, @grahamwhaley, @jcvenegas - what do you think?

File: docs/developers-clear-containers-install.md (hunk 2 above)
Comment (jodh-intel) [reply docs/developers-clear-containers-install.md]: On reflection, I think we can just drop that entire paragraph.

File: docs/developers-clear-containers-install.md (hunk 1 above)
Comment (jodh-intel) [reply docs/developers-clear-containers-install.md]: @sameo, @devimc, @grahamwhaley, @jcvenegas - I've added more details for the agent, including osbuilder details although it's debatable whether some of this should live in osbuilder's README instead. Thoughts?

File: docs/developers-clear-containers-install.md [hunk 5]
```diff
@@ -57,8 +58,10 @@ Both projects ship ``cc-proxy`` and they are not compatible with each other.
    ```bash
    $ go get -d github.com/clearcontainers/runtime
    $ go get -d github.com/clearcontainers/proxy
+   $ go get -d github.com/clearcontainers/agent
```
Comment (grahamwhaley) [inline docs/developers-clear-containers-install.md]: nitpick - you have Agent after Shim in the previous list - the opposite way around to here.

File: docs/developers-clear-containers-install.md (hunk 1 above)
Comment (jcvenegas) [reply docs/developers-clear-containers-install.md]: @jodh-intel nice!,  IMO it should be part of osbuilder  or an special section to build CC artifacts.

Comment (grahamwhaley) [reply docs/developers-clear-containers-install.md]: I suspect the info on how to install the agent is probably best placed in the osbuilder repo, and then we can reference that from both here and in the agent repo itself (the agent repo is somewhat lacking in any docs/details :-)
Having said that, for now we can put the info here and move it later to the osbuilder repo - no problem doing that if we need expediency.

File: docs/developers-clear-containers-install.md (hunk 5 above)
Comment (jodh-intel) [reply docs/developers-clear-containers-install.md]: Fixed.

**Comment by jodh-intel** [comment]
@grahamwhaley - can you give an ack/nack?

**Comment by grahamwhaley** [comment]
@jodh-intel You got a bunch of signal failures on semaphoreCI. Not clear if that is spurious or if you picked up something recently in a rebase maybe? :-(

**Comment by jodh-intel** [comment]
It seems happy now. I only changed a doc so this is either a latent bug or a semaphore issue, so yes, @chavafg please take a look at the failed build logs for this PR.
